collector/test/NetworkStatusNotifierTest.cpp

```diff
@@ -111,6 +126,80 @@ class MockNetworkConnectionInfoServiceComm : public INetworkConnectionInfoServic
   MOCK_METHOD(std::unique_ptr<IDuplexClientWriter<sensor::NetworkConnectionInfoMessage>>, PushNetworkConnectionInfoOpenStream, (std::function<void(const sensor::NetworkFlowsControlMessage*)> receive_func), (override));
 };
 
+/* gRPC payload objects are not strictly the ones of our internal model.
+   This helper class converts back NetworkConnectionInfoMessage into internal model in order
+   to compare them with the expected content. */
+class NetworkConnectionInfoMessageParser {
+ public:
+  NetworkConnectionInfoMessageParser(const sensor::NetworkConnectionInfoMessage& msg) {
+    for (auto conn : msg.info().updated_connections()) {
+      updated_connections_.emplace(
+          Connection(
+              conn.container_id(),
+              Endpoint(IPNetFromProto(conn.local_address()), conn.local_address().port()),
+              Endpoint(IPNetFromProto(conn.remote_address()), conn.remote_address().port()),
+              L4ProtocolFromProto(conn.protocol()),
+              conn.role() == sensor::ROLE_SERVER),
+          !conn.has_close_timestamp());
+    }
+  }
+
+  const std::unordered_map<Connection, bool, Hasher>& get_updated_connections() const { return updated_connections_; }
+
+ private:
+  std::unordered_map<Connection, bool, Hasher> updated_connections_;
```

Comment:
Could you pls add few words to clarify the content here? Mostly to mention that the boolean value indicates presence of "closed" timestamp.

Reply:
I added a few words. Do you think it makes it clear enough ?